pandas/core/indexes/base.py

```diff
@@ -270,7 +270,8 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
                                     **kwargs)
 
         # interval
-        if is_interval_dtype(data) or is_interval_dtype(dtype):
+        if ((is_interval_dtype(data) or is_interval_dtype(dtype)) and
```

Comment:
this hit in the tests?

Reply:
added a test for this that will appear in my next commit